### PR TITLE
feat: add infrastructure classes for hss system contract

### LINKED ISSUE #16520: [HIP 755] Implement needed infrastructure classes to define the `HssSystemContract`
Create `HssSystemContract`, `HssCallAttempt`, `ScheduleServiceEntity` and related classes to create the infrastructure for the HssSystemContract

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Add classes to define the Hedera Schedule Service system contract.

**Related issue(s)**:

Fixes #16520 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by david-bakin-sl — COMMENTED** [review]
First look - still have to look at unit tests.

**Review by david-bakin-sl — COMMENTED** [review]
LGTM.  Will approve when it moves out of draft

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/module-info.java [hunk 1]
```diff
@@ -28,8 +29,9 @@
     requires com.google.protobuf;
     requires org.apache.commons.lang3;
     requires org.bouncycastle.provider;
+    requires org.jetbrains.annotations; // javax.annotation.processing.Generated
```
[the remark this answers is not in the document]
Comment (lukelee-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/module-info.java:32]: removed

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM from `hedera-services` changes.

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Pull request merged**